lib/include/openamp/virtio.h

```diff
@@ -196,10 +201,99 @@ struct virtio_dispatch {
 	void (*notify)(struct virtqueue *vq);
 };
 
+/**
+ * @brief Create the virtio device virtqueue.
+ *
+ * @param vdev		Pointer to virtio device structure.
+ * @param flags		Create flag.
+ * @param nvqs		The virtqueue number.
+ * @param names		Virtqueue names.
+ * @param callbacks	Virtqueue callback functions.
+ *
+ * @return Pointer to virtio device structure.
+ */
 int virtio_create_virtqueues(struct virtio_device *vdev, unsigned int flags,
 			     unsigned int nvqs, const char *names[],
 			     vq_callback callbacks[]);
 
+/**
+ * @brief Delete the virtio device virtqueue.
+ *
+ * @param vdev	Pointer to virtio device structure.
+ *
+ * @return pointer to virtio device structure.
+ */
+#define virtio_delete_virtqueues(vdev) \
```

Comment:
Please use inline function instead. `checkpatch --strict` complains on this and followings macro.
`CHECK:MACRO_ARG_REUSE: Macro argument reuse 'vdev' - possible side-effects?`

Reply:
Done